### ISSUE TITLE
S3ObjectSummaryTable incorrectly displays keys with colons

### ISSUE THREAD

**Issue by CWindatt** [aws/aws-toolkit-eclipse#74]
I have an S3 table with keys formatted as date time 
certificate/5831565/RECEIPT/2017-07-14T13:49:45.298/s3.pdf

When this is displayed in the s3 bucket editor, the folder segment containing the datetime is truncated "45:45.298".  This is because the entry is being treated as a Path object and anything before the column is treated as a device.

com.amazonaws.eclipse.explorer.s3.S3ObjectSummaryTable.S3ObjectSummaryLabelProvider.getColumnText(Object, int)

**Comment by zhangzhx** [comment]
Thanks for reaching out to us. Can't repro in my s3 bucket editor, though. Can you show me a screencast for the incorrect display? Can you paste the error log if any?

**Notice, please erase all the confidential information in the screencast.**

Also, what's the toolkit's version you are using?

**Comment by CWindatt** [comment]
I'm working on a pull request.

**Comment by zhangzhx** [comment]
That's awesome! Thanks for the contribution :)

**Comment by CWindatt** [comment]
https://github.com/aws/aws-toolkit-eclipse/pull/75

**Comment by CWindatt** [comment]
Copying the comment here, my fix just replaces the colons with dashes.  Output isn't accurate, but at least it won't be truncated.  Also handles backslashes which Eclipse paths will not accept either.

**Comment by zhangzhx** [comment]
I'm worried about renaming the prefix for two reasons:
1. This misleads users with the wrong key name.
2. The name might be used in the toolkit as reference to the key to communicate with AWS. Renaming this incurs errors when used as key name to access AWS.

This seems only happening in Windows. I'm wondering if there is alternative way to resolve this.

**Comment by CWindatt** [comment]
I'm surprised this would only happen in Windows, as the Path object it creates is an Eclipse object that should always prevent inclusion of colons, slashes and backslashes.  This does not change the name used in the toolkit, only the path object that is used in the table contents.

I agree that this misleads users with the wrong key name, but to fix that would require a much more involved fix.  I'll take another look in case there is a way to escape the contents of a Path object.

**Comment by zhangzhx** [comment]
There is no such a problem on Mac, here is my screenshot:
![screen shot 2017-07-18 at 9 31 46 am](https://user-images.githubusercontent.com/7167513/28329402-8eed451e-6b9e-11e7-9514-7e61897a63b1.png)

I found the source code of the constructor of `Path`, it's removing the leading characters of ':' only in Windows.
```
	/** 
	 * Constructs a new path from the given string path.
	 * The string path must represent a valid file system path
	 * on the local file system. 
	 * The path is canonicalized and double slashes are removed
	 * except at the beginning. (to handle UNC paths). All forward
	 * slashes ('/') are treated as segment delimiters, and any
	 * segment and device delimiters for the local file system are
	 * also respected (such as colon (':') and backslash ('\') on some file systems).
	 *
	 * @param fullPath the string path
	 * @see #isValidPath(String)
	 */
	public Path(String fullPath) {
		String devicePart = null;
		if (WINDOWS) {
			//convert backslash to forward slash
			fullPath = fullPath.indexOf('\\') == -1 ? fullPath : fullPath.replace('\\', SEPARATOR);
			//extract device
			int i = fullPath.indexOf(DEVICE_SEPARATOR);
			if (i != -1) {
				//remove leading slash from device part to handle output of URL.getFile()
				int start = fullPath.charAt(0) == SEPARATOR ? 1 : 0;
				devicePart = fullPath.substring(start, i + 1);
				fullPath = fullPath.substring(i + 1, fullPath.length());
			}
		}
		initialize(devicePart, fullPath);
	}
```
TL;DR, but this might help
https://www.eclipse.org/eclipse/platform-core/documents/path_changes.html

**Comment by CWindatt** [comment]
```
objects[i++] = new Path(null, commonPrefix);
```
So if you explicitly tell the construction there is no device, it won't try to extract a device on Windows.  That sounds like a better solution.

**Comment by zhangzhx** [comment]
Sounds good. It's a much cleaner solution. Can you update the PR and I will merge it into mainline. This will be included in the next release.

**Comment by zhangzhx** [comment]
Fixed this in the [latest release v201708041710](https://github.com/aws/aws-toolkit-eclipse/releases/tag/v201708041710).
